source/common/http/http2/codec_impl.cc

```diff
@@ -357,6 +358,23 @@ void ConnectionImpl::StreamImpl::encodeMetadata(const MetadataMapVector& metadat
 void ConnectionImpl::StreamImpl::processBufferedData() {
   ENVOY_CONN_LOG(debug, "Stream {} processing buffered data.", parent_.connection_, stream_id_);
 
+  // Restore crash dump context when processing buffered data.
+  Event::Dispatcher& dispatcher = parent_.connection_.dispatcher();
+  ASSERT(dispatcher.trackedObjectStackIsEmpty());
```

Comment:
Might be nice to add a comment which explains why this is true. From the PR description, it sounds like this is guaranteed to be true because this method is called from a callback (and hence it's the root of the processing stack)?

Reply:
Makes sense. Can you add a comment this this effect?

Reply:
done